R/mock_httr_response.R

```diff
@@ -0,0 +1,30 @@
+#' Mocks an httr response
+#'
+#' @param status code integer.
+#' @param content list. 
+#' @param type character. json or text
+#' @return a response object.
+#' @export
+#' @examples
+#' mock_httr_response(200L, list(data = "it works!"), "json")
+mock_httr_response <- function(status_code, content, type) {
+  type <- match.arg(type, c("json", "text"))
+  if (identical(type, "json")) {
+    if(is.list(content)) { content <- jsonlite::toJSON(content, auto_unbox = TRUE) }
```

Comment:
json serialization in jsonlite can be a bit tricky
```r
toJSON(x, dataframe = c("rows", "columns", "values"), matrix = c("rowmajor",
       "columnmajor"), Date = c("ISO8601", "epoch"), POSIXt = c("string",
       "ISO8601", "epoch", "mongo"), factor = c("string", "integer"),
       complex = c("string", "list"), raw = c("base64", "hex", "mongo"),
       null = c("list", "null"), na = c("null", "string"), auto_unbox = FALSE,
       digits = 4, pretty = FALSE, force = FALSE, ...)
```

`digits` option always kills me - seems that truncating to 4 digits by default is a little bit to crude for most statistical applications, hence I usually set `digits = 10` in my serialization code.

```r
R-STABLE> a <- list(foo = 0.123456789)
R-STABLE> jsonlite::toJSON(a, auto_unbox = TRUE) %>% jsonlite::fromJSON(.) %>% all.equal(a, .)
[1] "Component “foo”: Mean relative difference: 0.0003500091"
```